### ISSUE TITLE
Failure/Error: Configuration.first.update(opts)

### ISSUE THREAD

**Issue by hennevogel** [openSUSE/open-build-service#11076]
I'm sure you have come across this. You try to run a spec and everything explodes because seemingly the database is not seeded and there isn't a `Configuration` anymore. You seed the test database and it's back to normal.

```bash
➜  api git:(bugfix/11035) docker-compose exec frontend /bin/bash -l
Directory: /obs/src/api
Mon May  3 16:29:59 UTC 2021
frontend@0ba58934f6f0:/obs/src/api> rspec spec/controllers/trigger_controller_spec.rb:220
warning: parser/current is loading parser/ruby25, which recognizes
warning: 2.5.9-compliant syntax, but you are running 2.5.8.
warning: please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.
warning: parser/current is loading parser/ruby25, which recognizes
warning: 2.5.9-compliant syntax, but you are running 2.5.8.
warning: please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.
Run options: include {:locations=>{"./spec/controllers/trigger_controller_spec.rb"=>[220]}}

Randomized with seed 36246

TriggerController
  #set_package
    multibuild package
      assigns flavor name (FAILED - 1)

Failures:

  1) TriggerController#set_package multibuild package assigns flavor name
     Failure/Error: Configuration.first.update(opts)
     
     NoMethodError:
       undefined method `update' for nil:NilClass
     # ./app/models/configuration.rb:73:in `update'
     # ./spec/support/database_cleaner.rb:41:in `block (2 levels) in <top (required)>'

Finished in 0.1381 seconds (files took 14.07 seconds to load)
1 example, 1 failure

Failed examples:

rspec ./spec/controllers/trigger_controller_spec.rb:224 # TriggerController#set_package multibuild package assigns flavor name

Randomized with seed 36246

frontend@0ba58934f6f0:/obs/src/api> rake db:seed RAILS_ENV=test
warning: parser/current is loading parser/ruby25, which recognizes
warning: 2.5.9-compliant syntax, but you are running 2.5.8.
warning: please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.
Seeding architectures table...
Seeding roles table...
Seeding users table...
Seeding roles_users table...
Seeding static_permissions table...
Seeding static permissions for admin role in roles_static_permissions table...
Seeding static permissions for maintainer role in roles_static_permissions table...
Seeding static permissions for reader role in roles_static_permissions table...
Seeding static permissions for downloader role in roles_static_permissions table...
Seeding attrib_namespaces table...
Seeding attrib_types table...
Seeding issue trackers ...
frontend@0ba58934f6f0:/obs/src/api> rspec spec/controllers/trigger_controller_spec.rb:220
warning: parser/current is loading parser/ruby25, which recognizes
warning: 2.5.9-compliant syntax, but you are running 2.5.8.
warning: please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.
Run options: include {:locations=>{"./spec/controllers/trigger_controller_spec.rb"=>[220]}}

Randomized with seed 5314

TriggerController
  #set_package
    multibuild package
      assigns flavor name

Finished in 1.12 seconds (files took 4.21 seconds to load)
1 example, 0 failures

Randomized with seed 5314

frontend@0ba58934f6f0:/obs/src/api>
```

Look at the `warning: parser/current is loading parser/ruby25, which recognizes` message when running the spec. In the case that fails, things are loaded twice. This consistently leads to the problem for me.

Why are things loaded twice?

**Comment by vpereira** [comment]
@hennevogel 

I don't see it loading things twice.. any other necessary step?

```
frontend@dd9fc8d2f021:/obs/src/api> rspec spec/controllers/trigger_controller_spec.rb:220
warning: parser/current is loading parser/ruby25, which recognizes
warning: 2.5.9-compliant syntax, but you are running 2.5.8.
warning: please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.
Run options: include {:locations=>{"./spec/controllers/trigger_controller_spec.rb"=>[220]}}

Randomized with seed 40798

TriggerController
  #set_package
    project with project-link and token that follows project-links
      assigns linked package

Finished in 2.33 seconds (files took 6.76 seconds to load)
1 example, 0 failures

Randomized with seed 40798

frontend@dd9fc8d2f021:/obs/src/api>     
```

**Comment by hennevogel** [comment]
It's not happening always.

**Comment by vpereira** [comment]
The issue with database cleaner was fixed here: #11265 

The other issues I wasn't able to reproduce @hennevogel

**Comment by hennevogel** [comment]
well maybe it's not an issue anymore after this. let's see...
